codec/reflectcodec/type_codec.go

```diff
@@ -318,20 +330,34 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 		p.PackBool(value.Bool())
 		return p.Err
 	case reflect.Ptr:
-		if value.IsNil() { // Can't marshal nil (except nil slices)
+		isNil := value.IsNil()
+		if nullable {
+			p.PackBool(isNil)
+			if isNil || p.Err != nil {
+				return p.Err
+			}
+		} else if isNil {
 			return errMarshalNil
 		}
-		return c.marshal(value.Elem(), p, c.maxSliceLen)
+
+		return c.marshal(value.Elem(), p, c.maxSliceLen, false)
```

Comment:
I have kind of a hard time reasoning about this -- why do we pass `false` here? Can `value.Elem()` never also be `nullable`?

Reply:
`nullable` is whatever we define it as... If the user wants `value.Elem()` to be nullable, then they should wrap it in a struct and explicitly mark it as `nullable`.